### PR TITLE
alsaLib: fix build under musl-libc

### PR DESCRIPTION
**Pull request by pjjw**
###### Motivation for this change
Presently, alsaLib doesn't build under musl. Looks like this was an inadvertent breakage from upstream, so pulling the upstream commit in to build under musl. Alpine linux takes the same approach.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pjjw** [comment]
@GrahamcOfBorg build pkgsMusl.alsaLib

File: pkgs/os-specific/linux/alsa-lib/default.nix [hunk 1]
```diff
@@ -11,6 +11,12 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./alsa-plugin-conf-multilib.patch
+    (fetchpatch {
+      # plucked from upstream master, delete in next release
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/alsa-lib/default.nix:15]: Please comment why we need this rather then from where you got it which we can easily see by the link.

Comment (pjjw) [reply pkgs/os-specific/linux/alsa-lib/default.nix:15]: fair point! added an additional comment about the patch's purpose. PTAL

**Pull request merged**